### ISSUE TITLE
[Feature request] FileTextSearcher - return text

### ISSUE THREAD

**Issue by markgibbons25** [VladPVS/FastSearchLibraryNETCore#1]
Thanks for open sourcing this!
Currently would need to re-read the whole file if we need to check the text of the matching file, which is inefficient I think, would be good to have an option to return the text in the result.

**Comment by VladPVS** [comment]
Hello!
First of all, thank you for your feedback.
It always pleasantly to realize that work was useful and it wasn't done in vain.
This library has been written only for research purposes and intended for distribution among anyone who needs it.
It was never used in any production, at least by me.

Regarding your suggestion, 
I was testing library on set of small text files (usually less than 100 Kb each),
therefore they were read from SSD almost immediately.

Probably, in some specific cases, apparently like yours, found files may be large and it makes sense to add option 
to return file content together with the its path.  

I could try to add the appropriate property to FileEventArgs class.
Thus, file content will be available throught event properties when one uses FileTextSearcher class.

What do you think about it?

**Comment by markgibbons25** [comment]
Yep that sounds good. I admit that this tool runs so fast that this wouldn't be any major performance improvement, and I suppose it could have a down side of more memory usage. At the moment I'm just using the FileSearcher and then reading the files in the FilesFound event, but my code isn't quite as multi-thread-optimized as yours is.

**Comment by VladPVS** [comment]
I added possibility to get file content with the file path, but since these changes cover only this particular case
and need much time for testing, so I suppose it's unjustified to make available this version for everyone. Could you give your email or please write me on putchenko.vs@gmail.com?
